datasketches-memory-java8/src/main/java/org/apache/datasketches/memory/internal/BaseWritableMemoryImpl.java

```diff
@@ -76,6 +76,9 @@ public static WritableMemory wrapHeapArray(final Object array, final long offset
     Util.negativeCheck(offsetBytes, "offsetBytes");
     Util.negativeCheck(lengthBytes, "lengthBytes");
     Objects.requireNonNull(byteOrder, "byteOrder must be non-null");
+    if (array instanceof byte[]) {
+      ResourceImpl.checkBounds(offsetBytes, lengthBytes, ((byte[])array).length);
+    }
```

Comment:
This fixes a subtle bug that is only encountered if the user tries to use the ds/internal/BaseWritableBufferImpl directly -- which is out-of-scope.